### ISSUE TITLE
waitTimeout does not throw exception

### ISSUE THREAD

**Issue by chrisgeo** [segmentio/nightmare#397]
I might be misunderstanding [this line](https://github.com/segmentio/nightmare/blob/master/lib/actions.js#L297), my apologies beforehand if my expectation is off:

```javascript
return done(new Error('.wait() timed out after '+self.optionWaitTimeout+'msec'));
```

This doesn't seem to throw an error/exception when using vo, example code below:

```javascript
var Nightmare = require('nightmare'),
    vo = require('vo');


vo(function* () {
  var nightmare = Nightmare({
    show: process.env.SHOW_WINDOW || false,
    width: 1305,
    height: 800,
    waitTimeout: 1000 // we aren't going to wait this long unfortunately
  }), url;

  nightmare.goto('http://www.cnn.com')
    .wait(function(){
      return document.querySelector('.some-fake-selector');
    })
    .wait(1000); // give it an extra second, arbitrary time.

  url = yield nightmare.url();

  yield nightmare.end();

  return url;

})(function (err, result) {
  console.log(result, err);
  if (err) {
    console.error(err);
    return err;
  }

  console.log('Found url=%s', result);
});

```

For some odd reason, the path to `url = yield nightmare.url()` gets called instead of an exception being thrown.

**Comment by yoz** [comment]
Are you missing a `yield` in front of the `nightmare.goto` line?

**Comment by chrisgeo** [comment]
@yoz The results are the same, a yield is only needed when you want to return the result. You could run the script in this incarnation as well and it will yield the same result:
```javascript
var Nightmare = require('nightmare'),
    vo = require('vo');


vo(function* () {
  var nightmare = Nightmare({
    show: process.env.SHOW_WINDOW || false,
    width: 1305,
    height: 800,
    waitTimeout: 1000 // we aren't going to wait this long unfortunately
  }), url;

  yield nightmare.goto('http://www.cnn.com')
    .wait(function(){
      return document.querySelector('.some-fake-selector');
    })
    .wait(100);

  url = yield nightmare.url();

  yield nightmare.end();

  return url;

})(function (err, result) {
  console.log(result, err);
  if (err) {
    console.error(err);
    process.exit(1);
  }

  console.log('Found url=%s', result);
});
```

Output:

```
http://www.cnn.com/ null
Found url=http://www.cnn.com/
```

Now, when you remove the ending `wait(100)`, it does throw the exception but the process hangs without exiting:
```
undefined [Error: .wait() timed out after 1000msec]
[Error: .wait() timed out after 1000msec]
```

**Comment by yuki-sato** [comment]
Hi. It also occurred to me.
And this seems Electron's issue or design.

waitfn() in Nightmare will perform dom search javascript every 250ms.
But suddenly javascript stop responding.

      parent.on('javascript', function(src) {
    renderer.once('response', function(event, response) {
      parent.emit('javascript', null, response);
    });

    renderer.once('error', function(event, error) {
      parent.emit('javascript', error);
    });

    renderer.once('log', function(event, args) {
      parent.emit.apply(parent, ['log'].concat(args));
    });

    win.webContents.executeJavaScript(src);
      });

This is not always happened but also specific page cause it.
when no javascript response come, i saw the logs that Electron dispatched messages

```
did-get-response-details
did-get-response-details
page-favicon-updated
did-stop-loading
did-frame-finish-load
did-get-response-details
did-get-response-details
```

Electron should dispatch "response" or "error". But after these message dispatched, never happened.
I guess this is electron's bug or design.

I'm keeping research what happened in electron.

If someone want to fix instantly. Just add long wait before .wait("#dom") to over that event
`.wait(10000)`
Or, use setTimeout your self.

**Comment by rosshinkley** [comment]
I believe this issue was fixed in #458, so I'm closing it.  If this is still a problem, feel free to reopen/open a new issue.
